### PR TITLE
feat: page-loading wrapper around io handle

### PR DESCRIPTION
**Pull request by rphmeier**
Note that the intended usage will be totally clear in future PRs. You can see that the API is well-suited for being called as part of a larger procedure that tracks the submitted page requests in e.g. a slab and applies completions.

### CODE REVIEW

**Comment by rphmeier** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/295?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#296** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/296?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#295** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/295?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#293** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/293?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#292** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/292?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#291** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/291?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rphmeier and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by gabriele-0201 — APPROVED** [review]

File: nomt/src/bitbox/mod.rs [hunk 1]
```diff
@@ -256,55 +256,205 @@ impl PageLoader {
...
+        self.user_data
+    }
+
+    pub fn apply_to(self, load: &mut PageLoad) -> Option<(Vec<u8>, BucketIndex)> {
+        assert!(load.needs_completion());
+        let PageLoadState::Submitted(bucket) = load.state else {
+            unreachable!()
+        };
+
+        if self.page[PAGE_SIZE - 32..] == load.page_id.encode() {
```
Comment (pepyakin) [inline nomt/src/bitbox/mod.rs:409]: nit: asks for a dedicated function?

File: nomt/src/bitbox/mod.rs [hunk 2]
```diff
@@ -256,55 +256,205 @@ impl PageLoader {
...
+                    })),
+                    _ => panic!(),
+                }
+            }
+            Err(TryRecvError::Empty) => Ok(None),
+            Err(TryRecvError::Disconnected) => anyhow::bail!("I/O pool hangup"),
+        }
+    }
+
+    /// Receive the next completion, blocking the current thread.
```
Comment (pepyakin) [inline nomt/src/bitbox/mod.rs:372]: Or blocking forever if in incorrect state, AFAIU.

**Review by pepyakin — APPROVED** [review]

File: nomt/src/bitbox/mod.rs (hunk 2 above)
Comment (rphmeier) [reply nomt/src/bitbox/mod.rs:372]: True

**Comment by pepyakin** [comment]
### Merge activity

* **Aug 23, 10:23 AM EDT**: @pepyakin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/295).

**Pull request merged**